src/common/utils/registry/registry.go

```diff
@@ -130,9 +128,18 @@ func (r *Registry) Catalog() ([]string, error) {
 	return repos, nil
 }
 
-// Ping ...
+// Ping checks by Head method
 func (r *Registry) Ping() error {
-	req, err := http.NewRequest(http.MethodHead, buildPingURL(r.Endpoint.String()), nil)
+	return r.ping(http.MethodHead)
```

Comment:
Duplicated changes? ecr adapter also has the same change.

Reply:
Yes, because this is two seperate PR, so I copy this code, in case they can't be merged together